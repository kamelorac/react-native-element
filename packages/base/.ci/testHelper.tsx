import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
