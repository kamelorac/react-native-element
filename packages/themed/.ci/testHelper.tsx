import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
