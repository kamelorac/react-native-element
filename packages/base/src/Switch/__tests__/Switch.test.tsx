import React from 'react';
import { Switch } from '../index';
import { renderWithWrapper, fireEvent } from '../../../.ci/testHelper';

describe('Switch Component', () => {
  it('should match snapshot', () => {
    const { toJSON } = renderWithWrapper(<Switch value color="green" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with value', () => {
    const { wrapper } = renderWithWrapper(
      <Switch color={'purple'} />,
      'RNE__SWITCH'
    );
    expect(wrapper.props).toMatchObject({
      onTintColor: 'purple',
    });
  });

  it('should have an onValueChange event', () => {
    const onValueChange = jest.fn();
    const { wrapper } = renderWithWrapper(
      <Switch value onValueChange={onValueChange} />,
      'RNE__SWITCH'
    );
    fireEvent(wrapper, 'onValueChange');
    expect(onValueChange).toHaveBeenCalledTimes(1);
  });

  it('should renders style', () => {
    const { wrapper } = renderWithWrapper(
      <Switch value color="green" style={{ margin: 10 }} />,
      'RNE__SWITCH'
    );
    expect(wrapper.props.style[1]).toMatchObject({
      margin: 10,
    });
  });

  it('should contain the required accessibility properties', () => {
    const enabledComponent = renderWithWrapper(<Switch value />);
    const enabledSwitch = enabledComponent.getByRole('switch');
    expect(enabledSwitch.props.accessibilityState).toMatchObject({
      checked: true,
      disabled: false,
    });

    const disabledComponent = renderWithWrapper(
      <Switch value={false} disabled />
    );
    const disabledSwitch = disabledComponent.getByRole('switch');
    expect(disabledSwitch.props.accessibilityState).toMatchObject({
      checked: false,
      disabled: true,
    });
  });
});
