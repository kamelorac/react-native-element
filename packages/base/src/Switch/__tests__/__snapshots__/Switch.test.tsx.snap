// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <RCTSwitch
    accessibilityRole="switch"
    accessibilityState={
      {
        "checked": true,
        "disabled": false,
      }
    }
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="green"
    style={
      {
        "height": 31,
        "width": 51,
      }
    }
    testID="RNE__SWITCH"
    tintColor=""
    value={true}
  />
</View>
`;
